src/dynamics/tests/initial_conditions/ic_baro_dry_jw06.F90

```diff
@@ -236,40 +243,63 @@ subroutine bc_dry_jw06_set_ic(vcoord, latvals, lonvals, U, V, T, PS, PHIS, &
         end if
       end if
       if (lq) then
+         !Get water vapor constituent index:
+         call const_get_index(wv_stdname, ix_q)
+
+         !Determine which "Q" variable index matches water vapor:
+         m_cnst_ix_q = findloc(m_cnst, ix_q, dim=1)
+
         do k = 1, nlev
           where(mask_use)
-            Q(:,k,1) = 0.0_r8
+            Q(:,k,m_cnst_ix_q) = 0.0_r8
           end where
         end do
-!Un-comment once constituents are working in CAMDEN -JN:
-#if 0
         if(masterproc.and. verbose_use) then
-          write(iulog,*) '         ', trim(cnst_name(m_cnst(1))), ' initialized by "',subname,'"'
+          write(iulog,*) '         ', wv_stdname, ' initialized by "',subname,'"'
         end if
-#endif
       end if
     end if
 
-!Un-comment once constituents are working in CAMDEN -JN:
-#if 0
-    if (lq) then
-      ncnst = size(m_cnst, 1)
-      if ((vcoord == vc_moist_pressure) .or. (vcoord == vc_dry_pressure)) then
-        do m = 2, ncnst
-          call cnst_init_default(m_cnst(m), latvals, lonvals, Q(:,:,m_cnst(m)),&
-               mask=mask_use, verbose=verbose_use, notfound=.false.)
-        end do
-      end if
-    end if
-#else
     if (lq) then
+      ncnst = size(m_cnst)
       if ((vcoord == vc_moist_pressure) .or. (vcoord == vc_dry_pressure)) then
-        !Initialize cloud liquid and rain until constituent routines are enabled:
-        Q(:,:,ix_cld_liq) = 0.0_r8
-        Q(:,:,ix_rain)    = 0.0_r8
-      end if
-    end if
-#endif
+        do m = 1, ncnst
+
+          !Skip water vapor, as it was aleady set above:
+          if (m == m_cnst_ix_q) cycle
+
+          !Extract constituent minimum value:
+          const_qmin_value = const_qmin(m_cnst(m))
+
+          !Initialize constituent to its minimum value:
+          Q(:,:,m) = real(const_qmin_value, r8)
+
+          if (iret /= 0) then
+            call endrun(errmsg, file=__FILE__, line=__LINE__)
+          end if
+
+          if (const_has_default) then
+
+            !If default value exists, then extract default value
+            !from constituent properties object:
+            call const_props(m_cnst(m))%default_value(const_default_value, &
```

Comment:
I think we're missing this line somewhere prior to this:

```
const_props => cam_model_const_properties()
```

Reply:
Great catch!  In fact it looks like I am missing way more than that (I am kinda surprised my original tests ran at all...).